hangman/lib/hangman/move.ex

```diff
@@ -0,0 +1,55 @@
+defmodule Hangman.Move do
+
+  # try to specify that guess should a String with length of one
+  # but guard clauses forbid invoking my functions or even String.length
+  def make_move(game = %Hangman.Struct{}, guess) when is_bitstring(guess) do
+    #once a movement is made
+    #fisr update used and last_guess
+    last_guess = guess
+    new_used   =
+      case guess in game.used do
+	true   -> game.used
+	false  -> game.used ++ List.wrap(guess)
+      end
+    new_game   = %Hangman.Struct{ game | used: new_used, last_guess: last_guess}
+    final_game = is_in(guess in new_game.word, new_game, guess)
+    { final_game, Hangman.Tally.tally(final_game) }
+  end
+
+  def make_move(_,_) do
+    # when fed invalid data
+    raise RuntimeError, message: "A Hangman.Struct and a letter are required here!"
+    { :error, "A Hangman.Struct and a letter are required here!" }
+  end
+  
+  def is_in(true, game, guess) do
```

Comment:
I strongly recommend you review the section about lists. 

You are writing code in a Java/C++ style, treating lists are if they were arrays. Not only is this a lot harder to do—it also leads to errors.

For example, your code here fails if a letter is duplicated in the word to be guessed: it only finds the first one.